src/super_gradients/common/factories/datasets_factory.py

```diff
@@ -0,0 +1,27 @@
+from super_gradients.common.factories.base_factory import BaseFactory
+from super_gradients.training.datasets.dataset_interfaces import TestDatasetInterface, \
+    LibraryDatasetInterface, \
+    ClassificationDatasetInterface, Cifar10DatasetInterface, Cifar100DatasetInterface, \
+    ImageNetDatasetInterface, TinyImageNetDatasetInterface, \
+    CoCoDetectionDatasetInterface, CoCoSegmentationDatasetInterface, CoCo2014DetectionDatasetInterface, \
+    PascalAUG2012SegmentationDataSetInterface, PascalVOC2012SegmentationDataSetInterface
+
+
+class DatasetsFactory(BaseFactory):
+
+    def __init__(self):
+        type_dict = {
+            "test_dataset": TestDatasetInterface,
+            "library_dataset": LibraryDatasetInterface,
+            "classification_dataset": ClassificationDatasetInterface,
+            "cifar_10": Cifar10DatasetInterface,
+            "cifar_100": Cifar100DatasetInterface,
+            "imagenet": ImageNetDatasetInterface,
+            "tiny_imagenet": TinyImageNetDatasetInterface,
+            "coco2017_detection": CoCoDetectionDatasetInterface,
+            "coco2014_detection": CoCo2014DetectionDatasetInterface,
+            "coco2017_segmentation": CoCoSegmentationDatasetInterface,
+            "pascal_voc": PascalVOC2012SegmentationDataSetInterface,
```

Comment:
We should state the exact dataset (i.e year, voc/aug, segmentation/detection) from pascal. Note that we also have PascalVOCUnifiedDetectionDataSetInterface

Reply:
fixed.
PascalVOCDetectionDataSet does not have an interface currently

Reply:
Sure it does, https://github.com/Deci-AI/super-gradients/blob/2aa4045d9236b45623de4d892dd7c767c8398566/src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py#L697

Reply:
Also cityscapes